src/kernel/emulation/linux/misc/sysctl_machdep.c

```diff
@@ -51,3 +80,139 @@ sysctl_handler(handle_brand_string)
 	return 0;
 }
 
+sysctl_handler(handle_max_basic)
+{
+	setup(0);
+
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+
+	sprintf(old,"%d",eax);
```

Comment:
This can cause actually cause segfaults because `sysctl` can be called with a `NULL` value for `old`, in order to determine the number of characters necessary. `copyout_string` already handles this, but doing `__simple_sprintf` (or `sprintf`) directly into the string can cause a segfault.

@bugaevc suggested that you could create a small stack-allocated buffer for the string, `__simple_sprintf` to that, and then use `copyout_string` on it. e.g.:
```suggestion
	char tmp[64]; // probably more than enough
	__simple_sprintf(tmp, "%d", eax);
	copyout_string(tmp, (char*)old, oldlen);
```